homeassistant/helpers/entity_component.py

```diff
@@ -229,15 +229,17 @@ def async_update_group(self):
 
         This method must be run in the event loop.
         """
+        def sorted_ids():
```

Comment:
We shouldn't instantiate a method on each call to this method. Instead just copy paste the method in both places.

You should also be able to do just this: `sorted(self.entities, key=lambda x: self.entities.name)`